src/jit/importer.cpp

```diff
@@ -13587,9 +13621,10 @@ void Compiler::impImportBlockCode(BasicBlock* block)
                     Verify(elemTypeHnd == nullptr ||
                                !(info.compCompHnd->getClassAttribs(elemTypeHnd) & CORINFO_FLG_CONTAINS_STACK_PTR),
                            "array of byref-like type");
-                    tiRetVal = verMakeTypeInfo(resolvedToken.hClass);
                 }
 
+                tiRetVal = verMakeTypeInfo(resolvedToken.hClass);
```

Comment:
We can use the stack type handle to set temp types; this makes the more precise type available more generally.